tf2onnx/optimizer/merge_duplicated_nodes_optimizer.py

```diff
@@ -90,3 +92,41 @@ def _skip_node_type(node):
     @staticmethod
     def _len_of_node_output(node):
         return len(node.output)
+
+    @staticmethod
+    def _are_attr_equal(a, b):
+        if len(a) != len(b):
+            return False
+        for k, v in a.items():
+            if v.type == onnx_pb.AttributeProto.FLOAT or \
+                    v.type == onnx_pb.AttributeProto.INT or \
+                    v.type == onnx_pb.AttributeProto.STRING or \
+                    v.type == onnx_pb.AttributeProto.FLOATS or \
+                    v.type == onnx_pb.AttributeProto.INTS or \
+                    v.type == onnx_pb.AttributeProto.STRINGS:
+                if v != b[k]:
+                    return False
+            elif v.type == onnx_pb.AttributeProto.TENSOR:
+                t1 = v.t
+                t2 = b[k].t
+                if not (t1.dims == t2.dims and t1.data_type == t2.data_type and t1.raw_data == t2.raw_data):
+                    return False
+            elif v.type == onnx_pb.AttributeProto.TENSORS:
+                if len(v.ts) != len(b[k].ts):
+                    return False
+                for i in range(len(v.ts)):
+                    t1 = v.ts[i]
+                    t2 = b[k].ts[i]
+                    if not (t1.dims == t2.dims and t1.data_type == t2.data_type and t1.raw_data == t2.raw_data):
+                        return False
+            elif v.type == onnx_pb.AttributeProto.GRAPH:
```

Comment:
if "node.attr == nodes_to_process[0].attr" is not work, can we compare graph directly?

Reply:
for the time being I am comparing graphs using the `==` since that's what we have had in the code since when this optimizer was written. However a more solid solution would be first to skip comparing graph's `name` ans `string_doc` fields and then compare nodes
I might take another stab at this

Reply:
@zhijxu-MS I elaborated on the solution. and moved the routines to utils.py for potential reusability. Please review when you get a chance.
Important changes are not relying on raw_data, making the overall flow a little more stable, and also adding comparison for graph attributes